### PR TITLE
Fix compute shape for the fused_concat

### PR DESCRIPTION
**Pull request by umangyadav**
Fixes compilation for the `migraphx-driver yolov4.onnx --fp16`
```
Running [ MIGraphX Version: 2.9.0.538ffcce ]: ./migraphx-driver perf yolov4.onnx --fp16
Compiling ...
Reading: yolov4.onnx
terminate called after throwing an instance of 'migraphx::version_2_9_0::exception'
  what():  /long_pathname_so_that_rpms_can_package_the_debug_info/src/extlibs/AMDMIGraphX/src/fuse_concat.cpp:54: compute_shape: FUSED_CONCAT: Missing fused modules

```

### CODE REVIEW

File: src/fuse_concat.cpp [hunk 1]
```diff
@@ -50,8 +50,12 @@ struct fused_concat
     shape compute_shape(std::vector<shape> inputs, const std::vector<module_ref>& mods) const
     {
         check_shapes{inputs, *this}.same_ndims();
-        if((inputs.size() + 1) == mods.size())
-            MIGRAPHX_THROW("FUSED_CONCAT: Missing fused modules");
+        // original concat can have multiple inputs. Let's say it has `n` input args.
+        // Each of those `n` input args are converted into pointwise modules that take atleast 1
+        // input parameter. Fused concat will have `n+1` module arguments. `n+1`th module is the
+        // post pointwise module which can take 0 or more input arguments.
+        if((inputs.size() + 1) < mods.size())
```
Comment (TedThemistokleous) [inline src/fuse_concat.cpp:57]: Your comment, does that imply that we should be checking <= instead of just < here?

Comment (umangyadav) [reply src/fuse_concat.cpp:57]: No, `==` is a valid. It needs to check if its' invalid and throw.

Comment (umangyadav) [reply src/fuse_concat.cpp:57]: added assert for the `input_iter`

**Review by TedThemistokleous — APPROVED** [review]

**Comment by TedThemistokleous** [comment]
@pfultz2 @CharlieL7  is there anything you guys have on this? This is coming up in Navi32 defect meeting

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**